enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisterAllocation.scala

```diff
@@ -141,6 +141,18 @@ object RegisterAllocation {
         result += (lp -> rhsPipeline)
         rhsPipeline
 
+      case SemiApply(lhs, rhs) =>
+        val lhsPipeline = allocate(lhs, nullable, argument)
+        val rhsPipeline = allocate(rhs, nullable, Some(lhsPipeline.deepClone()))
```

Comment:
I think we need it. The argument being sent in to the `rhs` will be used to "seed" the `rhs` pipeline, but other operators will add registers to the `rhs`-pipeline, and these new allocations should not show up on the `lhs`-pipeline.